pkg/sink/codec/common/message.go

```diff
@@ -120,3 +128,25 @@ func NewMsg(
 
 	return ret
 }
+
+// ClaimCheckMessage is the message sent to the claim-check external storage.
+type ClaimCheckMessage struct {
+	Key   []byte `json:"key"`
+	Value []byte `json:"value"`
+}
+
+// UnmarshalClaimCheckMessage unmarshal bytes to ClaimCheckMessage.
+func UnmarshalClaimCheckMessage(data []byte) (*ClaimCheckMessage, error) {
+	var m ClaimCheckMessage
+	err := json.Unmarshal(data, &m)
+	return &m, err
+}
+
+// NewClaimCheckFileName return file name for sent the message to claim check storage.
+func NewClaimCheckFileName(e *model.RowChangedEvent) string {
+	elements := []string{e.Table.Schema, e.Table.Table, strconv.FormatUint(e.CommitTs, 10)}
+	elements = append(elements, e.GetHandleKeyColumnValues()...)
```

Comment:
what if the file name is too long?  And seems in some cases the file name is not readable.

Reply:
It' rare to see file name too long (longer than 1024 bytes), if so use uuid to generate a random filename.

readability is not consider here, filename is only used to identify and fetch object